util/src/io/error.rs

```diff
@@ -0,0 +1,97 @@
+pub type Result<T> = core::result::Result<T, self::Error>;
+#[derive(Debug)]
+pub struct Error {
+    kind: ErrorKind,
+    details: Option<&'static str>,
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[non_exhaustive]
+pub enum ErrorKind {
```

Comment:
I'm guessing we aren't using most of these `ErrorKind` values right now, and it's possible some will never be used in a kernel (e.g. `PermissionDenied`).
It might be worth trimming down this list a bit to only contain values we need in a kernel, and add new ones as we need them, don't have strong feelings about that, though.

Reply:
I was going to do this initially, but I kept needing to add more error kinds. I am also anticipating we _might_ want to use mycelium_util as a replacement for std in non-kernel-mode components (microkernel servers?) until we can target std at mycelium's interface?